GVRf/Framework/src/org/gearvrf/scene_objects/GVRCylinderSceneObject.java

```diff
@@ -215,28 +216,32 @@ private void createCap(float radius, float height, int numSlices,
         }
     }
 
-    private void createBody(float bottomRadius, float topRadius, float height, int numStacks, int numSlices) {
+    private void createBody(float bottomRadius, float topRadius, float height,
+            int numStacks, int numSlices) {
         float difference = bottomRadius - topRadius;
         float halfHeight = height / 2.0f;
 
-        for(int stack = 0; stack < numStacks; stack++) {
-            float stackPercentage0 = ((float)(stack)/numStacks);
-            float stackPercentage1 = ((float)(stack+1)/numStacks);
+        for (int stack = 0; stack < numStacks; stack++) {
+
+            int initVertexCount = vertexCount;
+
+            float stackPercentage0 = ((float) (stack) / numStacks);
+            float stackPercentage1 = ((float) (stack + 1) / numStacks);
 
             float t0 = 1.0f - stackPercentage0;
             float t1 = 1.0f - stackPercentage1;
             float y0 = -halfHeight + (stackPercentage0 * height);
             float y1 = -halfHeight + (stackPercentage1 * height);
 
-            for(int slice = 0; slice < numSlices; slice++) {
-                float slicePercentage0 = ((float)(slice)/numSlices);
-                float slicePercentage1 = ((float)(slice+1)/numSlices);
-                float theta0 = slicePercentage0*2.0f*(float)Math.PI;
-                float theta1 = slicePercentage1*2.0f*(float)Math.PI;
-                float cosTheta0 = (float)Math.cos(theta0);
-                float sinTheta0 = (float)Math.sin(theta0);
-                float cosTheta1 = (float)Math.cos(theta1);
-                float sinTheta1 = (float)Math.sin(theta1);
+            for (int slice = 0; slice < numSlices; slice++) {
+                float slicePercentage0 = ((float) (slice) / numSlices);
+                float slicePercentage1 = ((float) (slice + 1) / numSlices);
+                float theta0 = slicePercentage0 * 2.0f * (float) Math.PI;
```

Comment:
There are two things that are less than ideal, here:

1. You're doing a (float) Math.PI 2*numSlices times.
2. You're taking `double` trig values, throwing away the low bits, then calculating. It would probably be better to do the math using `double` precision, and only drop down to `float` at the end.